core/src/main/scala/java/text/DecimalFormat.scala

```diff
@@ -1,57 +1,521 @@
 package java.text
 
-class DecimalFormat(private[this] val pattern: String, private[this] var symbols: DecimalFormatSymbols) extends Format {
+import java.math.{RoundingMode, BigDecimal => JavaBigDecimal, BigInteger => JavaBigInteger}
+import java.util.{Currency, Locale}
+import locales.{DecimalFormatUtil, LocaleRegistry, ParsedPattern}
+import scala.math.{max, min}
+
+// The constructor needs a non-localized pattern
+class DecimalFormat(private[this] val pattern: String, private[this] var symbols: DecimalFormatSymbols)
+    extends NumberFormat {
+
   def this(pattern: String) = this(pattern, DecimalFormatSymbols.getInstance())
 
-  //def this() = this("???", DecimalFormatSymbols.getInstance())
+  def this() = {
+    this(
+      LocaleRegistry.ldml(Locale.getDefault).flatMap{ _.numberPatterns.decimalPattern }.getOrElse(???),
```

Comment:
Right, it shouldn't be called ever but I'm not too happy about throwing an exception sincethe JVM never does that. We could extract the default pattern in the JVM with locale english and hardcode it there